torch/nn/functional.py

```diff
@@ -16,6 +16,43 @@
 from . import grad
 
 
+class Reduction:
```

Comment:
Is this intended to be part of public API?

Reply:
Please also add comments that tells people that this and the c++ side enums should be in sync

Reply:
Originally it was going to be public, but I think i'll make it private now that we're using strings.

Reply:
Should this remain public since I'm using it in other files?

Reply:
@li-roy Please don't. Just prefixing it with `_`. should be fine :)